src/sandbox/components/transactions-panel.tsx

```diff
@@ -0,0 +1,468 @@
+                  </Box>
+                }
+              >
+                <TxDetails
+                  txId={tx.tx_id}
+                  type={tx.tx_type}
+                  status={tx.tx_status}
+                  contractId={
+                    tx.tx_type === 'smart_contract'
+                      ? tx.smart_contract.contract_id
+                      : tx.contract_call.contract_id
+                  }
+                />
+              </React.Suspense>
+            </Box>
+          </Box>
+        ) : null}
+      </Box>
+    );
+  }
+);
+
+const TxList: React.FC = React.memo(() => {
+  const filters = useRecoilValue(filterState('sandbox'));
+  const { transactions, pendingTransactions, principal } = useUser({ suspense: true });
+
+  const filteredTxs = transactions?.filter(tx => filters.types.find(type => type === tx.tx_type));
+  const hasTxButIsFiltered = transactions?.length && filteredTxs?.length === 0;
+
+  const pendingList = React.useMemo(
+    () =>
+      pendingTransactions?.map(tx => (
+        <Flex borderBottom={border()} px="loose" alignItems="center" justifyContent="space-between">
+          <TxItem
+            hideRightElements
+            minimal
+            principal={principal}
+            tx={tx}
+            key={tx.tx_id}
+            width="auto"
+          />
+          <TxLink txid={tx.tx_id}>
+            <IconButton as="a" target="_blank" flexShrink={0} dark icon={ExternalLinkIcon} />
+          </TxLink>
+        </Flex>
+      )),
+    [pendingTransactions]
+  );
+
+  const txList = React.useMemo(
+    () =>
+      filteredTxs.map((tx, key, arr) =>
+        tx.tx_status !== 'success' && !filters.showFailed ? null : (
+          <SandboxTxItem key={key} tx={tx} key={tx.tx_id} isLast={key === arr.length - 1} />
+        )
+      ),
+    [filteredTxs, filters.types, filters.showFailed, transactions]
+  );
+
+  return (
+    <>
+      {pendingTransactions?.length && filters.showPending ? pendingList : null}
+      {filteredTxs?.length ? (
+        txList
+      ) : hasTxButIsFiltered ? (
+        <FilteredMessage filterKey="sandbox" />
+      ) : (
+        <Flex flexGrow={1} flexDirection="column" alignItems="center" justifyContent="center">
+          <Stack textAlign="center">
+            <Title>No Transactions</Title>
+            <Caption>Your list of transactions will display here.</Caption>
+          </Stack>
+        </Flex>
+      )}
+    </>
+  );
+});
+
+const TxLoadingPanel = React.memo(() => (
+  <Box p="extra-loose" flexGrow={1} width="100%">
+    <Flex flexDirection="column" alignItems="center">
+      <Grid placeItems="center" borderRadius="100%" size="64px" border={border()} boxShadow="mid">
+        <Pending opacity={0.5} size="32px" />
+      </Grid>
+      <Text mt="base" fontWeight="500" fontSize="14px" color={color('text-caption')}>
+        Fetching transactions...
+      </Text>
+    </Flex>
+  </Box>
+));
+
+export const TransactionsPanel = React.memo(props => {
```

Comment:
Maybe intended, but I always see my pending TX up top, even when I've filtered for other tx types